src/watertap_contrib/seto/property_models/basic_water_properties.py

```diff
@@ -0,0 +1,397 @@
+    def _flow_mass_comp(self):
+
+        self.flow_mass_comp = Var(
+            self.params.component_list,
+            initialize=1,
+            domain=PositiveReals,
+            doc="Mass flowrate of each component",
+            units=pyunits.kg / pyunits.s,
+        )
+
+        def rule_flow_mass_comp(b, j):
+            if j == "H2O":
+                return b.flow_mass_comp[j] == b.flow_vol * b.dens_mass
+            else:
+                return b.flow_mass_comp[j] == b.flow_vol * b.conc_mass_comp[j]
+
+        self.eq_flow_mass_comp = Constraint(
+            self.params.component_list, rule=rule_flow_mass_comp
+        )
+
+    def _temperature(self):
+        self.temperature = Var(
+            initialize=298.15,
+            bounds=(273.15, 373.15),
+            units=pyunits.K,
+            doc="Temperature",
+        )
+
+    def _pressure(self):
+        self.pressure = Var(
+            initialize=101325,
+            bounds=(1e5, None),
+            units=pyunits.Pa,
+            doc="Pressure",
+        )
+
+    def _dens_mass(self):
+        add_object_reference(self, "dens_mass", self.params.dens_mass)
+
+    def _visc_d(self):
+        add_object_reference(self, "visc_d", self.params.visc_d)
+
+    def get_material_flow_terms(self, j):
+        return self.flow_mass_comp[j]
+
+    def get_enthalpy_flow_terms(self, p):
+        raise NotImplementedError
+
+    def get_material_density_terms(self, j):
+        if j == "H2O":
+            return self.dens_mass
+        else:
+            return self.conc_mass_comp[j]
+
+    def get_energy_density_terms(self, p):
+        raise NotImplementedError
+
+    def default_material_balance_type(self):
+        return MaterialBalanceType.componentTotal
+
+    def default_energy_balance_type(self):
+        return EnergyBalanceType.none
+
+    def define_state_vars(self):
+        return {"flow_vol": self.flow_vol, "conc_mass_comp": self.conc_mass_comp}
+
+    def define_display_vars(self):
+        return {
+            "Volumetric Flowrate": self.flow_vol,
+            "Mass Concentration": self.conc_mass_comp,
+            "Temperature": self.temperature,
+        }
+
+    def get_material_flow_basis(self):
+        return MaterialFlowBasis.mass
+
+    def calculate_scaling_factors(self):
+
+        super().calculate_scaling_factors()
+
+        if iscale.get_scaling_factor(self.flow_vol) is None:
+            sf_Q = iscale.get_scaling_factor(self.flow_vol, default=1, warning=True)
+            iscale.set_scaling_factor(self.flow_vol, sf_Q)
+
+        for j, v in self.conc_mass_comp.items():
+            sf_c = iscale.get_scaling_factor(self.conc_mass_comp[j])
+            if sf_c is None:
+                try:
+                    sf_c = self.params.default_scaling_factor[("conc_mass_comp", j)]
+                except KeyError:
+                    iscale.get_scaling_factor(
+                        self.conc_mass_comp[j], default=1, warning=True
+                    )
```

Comment:
Is here missing a line to assign the scaling factor to `self.conc_mass_comp[j]` ?
Also I think it's better to also use the multiplicative inverse of the values as the default scaling factors, instead of 1, because as tested in the test file, the concentration could vary a lot among different solutes.

Reply:
Yes that is correct! Good catch, should be ``set_scaling_factor`` rather than ``get_scaling_factor``